riposte-core/src/main/java/com/nike/riposte/server/handler/RequestInfoSetterHandler.java

```diff
@@ -47,13 +59,31 @@ else if (msg instanceof HttpContent) {
                                                     + "HttpProcessingState. This should be impossible");
                 }
 
-                requestInfo.addContentChunk((HttpContent) msg);
+                int currentRequestLengthInBytes = requestInfo.addContentChunk((HttpContent) msg);
+                int configuredMaxRequestSize = getConfiguredMaxRequestSize(state);
+                
+                if (!isMaxRequestSizeValidationDisabled(configuredMaxRequestSize) && currentRequestLengthInBytes > configuredMaxRequestSize) {
+                    throw new TooLongFrameException("Request raw content length exceeded configured max request size of " + configuredMaxRequestSize);
+                }
             }
         }
 
         return PipelineContinuationBehavior.CONTINUE;
     }
 
+    private boolean isMaxRequestSizeValidationDisabled(int configuredMaxRequestSize) {
+        return configuredMaxRequestSize <= 0;
+    }
+
+    private int getConfiguredMaxRequestSize(HttpProcessingState state) {
+        int maxConfiguredRequestSize = maxRequestSizeInBytes;
+        Optional<Integer> endpointMaxRequestSizeOverride = state.getEndpointForExecution().maxRequestSizeInBytesOverride();
```

Comment:
Now that I think about it the endpoint could also return null. It would be a bad/broken endpoint but we still shouldn't NPE if someone does it. If null is returned from an endpoint's `maxRequestSizeInBytesOverride()` it should also be treated as an empty optional (i.e. use global default).